src/electionguard/guardian.py

```diff
@@ -163,18 +169,82 @@ def __init__(
         :param sequence_order: a unique number in [1, 256) that identifies this guardian
         :param number_of_guardians: the total number of guardians that will participate in the election
         :param quorum: the count of guardians necessary to decrypt
-        :param nonce_seed: an optional `ElementModQ` value that can be used to generate the `ElectionKeyPair`.
-                           It is recommended to only use this field for testing.
+        :param election_keys the private keys the guardian generated during a key ceremony
+        :param election_public_keys the public keys the guardian generated during a key ceremony
+        :param partial_key_backups the partial key backups the guardian generated during a key ceremony
         """
         self.id = id
         self.sequence_order = sequence_order
         self.set_ceremony_details(number_of_guardians, quorum)
-        self._backups_to_share = {}
-        self._guardian_election_public_keys = {}
-        self._guardian_election_partial_key_backups = {}
-        self._guardian_election_partial_key_verifications = {}
+        self._backups_to_share = {} if backups_to_share is None else backups_to_share
+        self._guardian_election_public_keys = (
+            {} if election_public_keys is None else election_public_keys
+        )
+        self._guardian_election_partial_key_backups = (
+            {} if partial_key_backups is None else partial_key_backups
+        )
+        self._guardian_election_partial_key_verifications = (
+            {}
+            if guardian_election_partial_key_verifications is None
+            else guardian_election_partial_key_verifications
+        )
+
+        if not election_keys is None:
+            self._election_keys = election_keys
+
+    @classmethod
+    def from_context_info(
+        cls,
+        id: str,
+        sequence_order: int,
+        number_of_guardians: int,
+        quorum: int,
+    ) -> "Guardian":
+        guardian = cls(id, sequence_order, number_of_guardians, quorum)
+        guardian.generate_election_key_pair(None)
+        return guardian
+
+    @classmethod
+    def from_nonce(
+        cls,
+        id: str,
+        sequence_order: int,
+        number_of_guardians: int,
+        quorum: int,
+        nonce_seed: Optional[ElementModQ] = None,
+    ) -> "Guardian":
+        """Creates a guardian with an `ElementModQ` value that will be used to generate
+        the `ElectionKeyPair`. This method should generally only be used for testing."""
+
+        guardian = cls(
+            id,
+            sequence_order,
+            number_of_guardians,
+            quorum,
+        )
+        guardian.generate_election_key_pair(nonce_seed)
```

Comment:
No reason to use this method forcing construction immediately. Directly, use the `generate_election_key_pair` function and in addition, remove the original function.